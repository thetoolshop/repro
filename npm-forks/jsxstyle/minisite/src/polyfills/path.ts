const extnameRegex = /(\.[a-z]+)$/i;

export const resolve = (path: string) => path;
export const isAbsolute = (path: string) => path.trim().startsWith('/');
export const dirname = (path: string) => path.split('/').slice(0, -1).join('/');
export const extname = (path: string) => extnameRegex.exec(path)?.[1] || '';
export const join = (...paths: string[]) => paths.join('/');
export const basename = (path: string) => path.split('/').pop() || path;
export const relative = (...paths: string[]) => paths.pop();
