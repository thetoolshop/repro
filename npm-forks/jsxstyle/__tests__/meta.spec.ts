import packlist from 'npm-packlist';
import * as path from 'path';
import glob from 'glob';
import { execSync } from 'child_process';
import { getPackages } from '@manypkg/get-packages';

const JSXSTYLE_ROOT = path.resolve(__dirname, '..');

describe('npm publish', () => {
  it('only publishes the intended files', async () => {
    const packages = await getPackages(JSXSTYLE_ROOT);
    const packagePromises = packages.packages
      // exclude private packages
      .filter((pkg) => !pkg.packageJson.private)
      .sort((a, b) => a.packageJson.name.localeCompare(b.packageJson.name))
      .map((pkg) =>
        // fetch file list and format it into something
        packlist({ path: pkg.dir }).then(
          (fileList) => `
${pkg.packageJson.name}
${pkg.packageJson.name.replace(/./g, '=')}
${fileList
  .map((fileName) => {
    // chunk filenames contain a hash that will change across builds
    const updatedFileName = fileName.replace(
      /(\.)[a-f0-9]{8}(\.c?js(?:\.map)?)$/,
      '$1[hash]$2'
    );
    return `- ${updatedFileName}`;
  })
  .sort()
  .join('\n')}
`
        )
      );

    await expect(Promise.all(packagePromises)).resolves.toMatchSnapshot();
  });
});

describe.skip('examples', () => {
  const exampleDir = path.resolve(__dirname, '..', 'examples');
  const examples = glob.sync('jsxstyle-*-example', { cwd: exampleDir });

  for (const example of examples) {
    it(`\`${example}\` builds correctly`, async () => {
      const cwd = path.join(exampleDir, example);
      expect.assertions(1);
      return expect(() =>
        execSync('npm run build', { cwd, stdio: 'inherit' })
      ).not.toThrow();
    }, 30000);
  }
});
