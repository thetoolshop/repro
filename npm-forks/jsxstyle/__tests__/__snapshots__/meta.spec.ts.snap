// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm publish only publishes the intended files 1`] = `
[
  "
jsxstyle
========
- LICENSE
- README.md
- lib/chunks/createClassNameGetter.[hash].cjs
- lib/chunks/createClassNameGetter.[hash].cjs.map
- lib/chunks/createClassNameGetter.[hash].js
- lib/chunks/createClassNameGetter.[hash].js.map
- lib/chunks/generateCustomPropertiesFromVariants.[hash].cjs
- lib/chunks/generateCustomPropertiesFromVariants.[hash].cjs.map
- lib/chunks/generateCustomPropertiesFromVariants.[hash].js
- lib/chunks/generateCustomPropertiesFromVariants.[hash].js.map
- lib/chunks/getExportsFromModuleSource.[hash].cjs
- lib/chunks/getExportsFromModuleSource.[hash].cjs.map
- lib/chunks/getExportsFromModuleSource.[hash].js
- lib/chunks/getExportsFromModuleSource.[hash].js.map
- lib/chunks/getStyleCache.[hash].cjs
- lib/chunks/getStyleCache.[hash].cjs.map
- lib/chunks/getStyleCache.[hash].js
- lib/chunks/getStyleCache.[hash].js.map
- lib/chunks/makeCssFunction.[hash].cjs
- lib/chunks/makeCssFunction.[hash].cjs.map
- lib/chunks/makeCssFunction.[hash].js
- lib/chunks/makeCssFunction.[hash].js.map
- lib/chunks/makeCustomProperties.[hash].cjs
- lib/chunks/makeCustomProperties.[hash].cjs.map
- lib/chunks/makeCustomProperties.[hash].js
- lib/chunks/makeCustomProperties.[hash].js.map
- lib/jsxstyle-bundler-utils/src/ModuleCache.d.ts
- lib/jsxstyle-bundler-utils/src/ModuleCache.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/babelUtils.d.ts
- lib/jsxstyle-bundler-utils/src/ast/babelUtils.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/evaluateAstNode.d.ts
- lib/jsxstyle-bundler-utils/src/ast/evaluateAstNode.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/evaluateAttributes.d.ts
- lib/jsxstyle-bundler-utils/src/ast/evaluateAttributes.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.cjs
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.cjs.map
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.d.ts
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.js
- lib/jsxstyle-bundler-utils/src/ast/extractStyles.js.map
- lib/jsxstyle-bundler-utils/src/ast/flattenSpreadAttributes.d.ts
- lib/jsxstyle-bundler-utils/src/ast/flattenSpreadAttributes.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/generatedUid.d.ts
- lib/jsxstyle-bundler-utils/src/ast/generatedUid.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getEvaluateAstNodeWithScopeFunction.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getEvaluateAstNodeWithScopeFunction.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getImportForSource.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getImportForSource.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getInlineImportString.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getInlineImportString.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getJsxAttribute.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getJsxAttribute.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getObjectProperty.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getObjectProperty.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getSourceModule.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getSourceModule.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getStaticBindingsForScope.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getStaticBindingsForScope.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/getValue.d.ts
- lib/jsxstyle-bundler-utils/src/ast/getValue.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/handleCssFunction.d.ts
- lib/jsxstyle-bundler-utils/src/ast/handleCssFunction.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/handleJsxAttributes.d.ts
- lib/jsxstyle-bundler-utils/src/ast/handleJsxAttributes.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/joinStringExpressions.d.ts
- lib/jsxstyle-bundler-utils/src/ast/joinStringExpressions.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/normalizeTernary.d.ts
- lib/jsxstyle-bundler-utils/src/ast/normalizeTernary.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/primitiveValueToNode.d.ts
- lib/jsxstyle-bundler-utils/src/ast/primitiveValueToNode.d.ts.map
- lib/jsxstyle-bundler-utils/src/ast/styleObjectUtils.d.ts
- lib/jsxstyle-bundler-utils/src/ast/styleObjectUtils.d.ts.map
- lib/jsxstyle-bundler-utils/src/getExportsFromModuleSource.d.ts
- lib/jsxstyle-bundler-utils/src/getExportsFromModuleSource.d.ts.map
- lib/jsxstyle-bundler-utils/src/index.cjs
- lib/jsxstyle-bundler-utils/src/index.cjs.map
- lib/jsxstyle-bundler-utils/src/index.d.ts
- lib/jsxstyle-bundler-utils/src/index.d.ts.map
- lib/jsxstyle-bundler-utils/src/index.js
- lib/jsxstyle-bundler-utils/src/index.js.map
- lib/jsxstyle-bundler-utils/src/makePromise.d.ts
- lib/jsxstyle-bundler-utils/src/makePromise.d.ts.map
- lib/jsxstyle-bundler-utils/src/noop.cjs
- lib/jsxstyle-bundler-utils/src/noop.cjs.map
- lib/jsxstyle-bundler-utils/src/noop.d.ts
- lib/jsxstyle-bundler-utils/src/noop.d.ts.map
- lib/jsxstyle-bundler-utils/src/noop.js
- lib/jsxstyle-bundler-utils/src/noop.js.map
- lib/jsxstyle-bundler-utils/src/typePredicates.d.ts
- lib/jsxstyle-bundler-utils/src/typePredicates.d.ts.map
- lib/jsxstyle-bundler-utils/src/types.d.ts
- lib/jsxstyle-bundler-utils/src/types.d.ts.map
- lib/jsxstyle-bundler-utils/src/wrapFileSystem.d.ts
- lib/jsxstyle-bundler-utils/src/wrapFileSystem.d.ts.map
- lib/jsxstyle-nextjs-plugin/src/index.cjs
- lib/jsxstyle-nextjs-plugin/src/index.cjs.map
- lib/jsxstyle-nextjs-plugin/src/index.d.ts
- lib/jsxstyle-nextjs-plugin/src/index.d.ts.map
- lib/jsxstyle-nextjs-plugin/src/index.js
- lib/jsxstyle-nextjs-plugin/src/index.js.map
- lib/jsxstyle-preact/src/componentFactory.d.ts
- lib/jsxstyle-preact/src/componentFactory.d.ts.map
- lib/jsxstyle-preact/src/index.cjs
- lib/jsxstyle-preact/src/index.cjs.map
- lib/jsxstyle-preact/src/index.d.ts
- lib/jsxstyle-preact/src/index.d.ts.map
- lib/jsxstyle-preact/src/index.js
- lib/jsxstyle-preact/src/index.js.map
- lib/jsxstyle-preact/src/styleCache.d.ts
- lib/jsxstyle-preact/src/styleCache.d.ts.map
- lib/jsxstyle-preact/src/types.d.ts
- lib/jsxstyle-preact/src/types.d.ts.map
- lib/jsxstyle-preact/src/useMatchMedia.d.ts
- lib/jsxstyle-preact/src/useMatchMedia.d.ts.map
- lib/jsxstyle-react/src/componentFactory.d.ts
- lib/jsxstyle-react/src/componentFactory.d.ts.map
- lib/jsxstyle-react/src/index.cjs
- lib/jsxstyle-react/src/index.cjs.map
- lib/jsxstyle-react/src/index.d.ts
- lib/jsxstyle-react/src/index.d.ts.map
- lib/jsxstyle-react/src/index.js
- lib/jsxstyle-react/src/index.js.map
- lib/jsxstyle-react/src/styleCache.d.ts
- lib/jsxstyle-react/src/styleCache.d.ts.map
- lib/jsxstyle-react/src/types.d.ts
- lib/jsxstyle-react/src/types.d.ts.map
- lib/jsxstyle-react/src/useMatchMedia.d.ts
- lib/jsxstyle-react/src/useMatchMedia.d.ts.map
- lib/jsxstyle-solid/src/componentFactory.d.ts
- lib/jsxstyle-solid/src/componentFactory.d.ts.map
- lib/jsxstyle-solid/src/index.cjs
- lib/jsxstyle-solid/src/index.cjs.map
- lib/jsxstyle-solid/src/index.d.ts
- lib/jsxstyle-solid/src/index.d.ts.map
- lib/jsxstyle-solid/src/index.js
- lib/jsxstyle-solid/src/index.js.map
- lib/jsxstyle-solid/src/styleCache.d.ts
- lib/jsxstyle-solid/src/styleCache.d.ts.map
- lib/jsxstyle-solid/src/types.d.ts
- lib/jsxstyle-solid/src/types.d.ts.map
- lib/jsxstyle-utils/src/addStyleToHead.d.ts
- lib/jsxstyle-utils/src/addStyleToHead.d.ts.map
- lib/jsxstyle-utils/src/componentStyles.d.ts
- lib/jsxstyle-utils/src/componentStyles.d.ts.map
- lib/jsxstyle-utils/src/createClassNameGetter.d.ts
- lib/jsxstyle-utils/src/createClassNameGetter.d.ts.map
- lib/jsxstyle-utils/src/dangerousStyleValue.d.ts
- lib/jsxstyle-utils/src/dangerousStyleValue.d.ts.map
- lib/jsxstyle-utils/src/generateCustomPropertiesFromVariants.d.ts
- lib/jsxstyle-utils/src/generateCustomPropertiesFromVariants.d.ts.map
- lib/jsxstyle-utils/src/getStringHash.d.ts
- lib/jsxstyle-utils/src/getStringHash.d.ts.map
- lib/jsxstyle-utils/src/getStyleCache.d.ts
- lib/jsxstyle-utils/src/getStyleCache.d.ts.map
- lib/jsxstyle-utils/src/hyphenateStyleName.d.ts
- lib/jsxstyle-utils/src/hyphenateStyleName.d.ts.map
- lib/jsxstyle-utils/src/index.cjs
- lib/jsxstyle-utils/src/index.cjs.map
- lib/jsxstyle-utils/src/index.d.ts
- lib/jsxstyle-utils/src/index.d.ts.map
- lib/jsxstyle-utils/src/index.js
- lib/jsxstyle-utils/src/index.js.map
- lib/jsxstyle-utils/src/makeCssFunction.d.ts
- lib/jsxstyle-utils/src/makeCssFunction.d.ts.map
- lib/jsxstyle-utils/src/makeCustomProperties.d.ts
- lib/jsxstyle-utils/src/makeCustomProperties.d.ts.map
- lib/jsxstyle-utils/src/parseStyleProps.d.ts
- lib/jsxstyle-utils/src/parseStyleProps.d.ts.map
- lib/jsxstyle-utils/src/processProps.d.ts
- lib/jsxstyle-utils/src/processProps.d.ts.map
- lib/jsxstyle-utils/src/stringHash.d.ts
- lib/jsxstyle-utils/src/stringHash.d.ts.map
- lib/jsxstyle-utils/src/types.d.ts
- lib/jsxstyle-utils/src/types.d.ts.map
- lib/jsxstyle-vite-plugin/src/index.cjs
- lib/jsxstyle-vite-plugin/src/index.cjs.map
- lib/jsxstyle-vite-plugin/src/index.d.ts
- lib/jsxstyle-vite-plugin/src/index.d.ts.map
- lib/jsxstyle-vite-plugin/src/index.js
- lib/jsxstyle-vite-plugin/src/index.js.map
- lib/jsxstyle-webpack-plugin/src/base64Loader.cjs
- lib/jsxstyle-webpack-plugin/src/base64Loader.cjs.map
- lib/jsxstyle-webpack-plugin/src/base64Loader.d.ts
- lib/jsxstyle-webpack-plugin/src/base64Loader.d.ts.map
- lib/jsxstyle-webpack-plugin/src/base64Loader.js
- lib/jsxstyle-webpack-plugin/src/base64Loader.js.map
- lib/jsxstyle-webpack-plugin/src/loader.cjs
- lib/jsxstyle-webpack-plugin/src/loader.cjs.map
- lib/jsxstyle-webpack-plugin/src/loader.d.ts
- lib/jsxstyle-webpack-plugin/src/loader.d.ts.map
- lib/jsxstyle-webpack-plugin/src/loader.js
- lib/jsxstyle-webpack-plugin/src/loader.js.map
- lib/jsxstyle-webpack-plugin/src/plugin.cjs
- lib/jsxstyle-webpack-plugin/src/plugin.cjs.map
- lib/jsxstyle-webpack-plugin/src/plugin.d.ts
- lib/jsxstyle-webpack-plugin/src/plugin.d.ts.map
- lib/jsxstyle-webpack-plugin/src/plugin.js
- lib/jsxstyle-webpack-plugin/src/plugin.js.map
- lib/jsxstyle-webpack-plugin/src/types.d.ts
- lib/jsxstyle-webpack-plugin/src/types.d.ts.map
- package.json
",
]
`;
